core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -526,14 +532,17 @@ class ReplicaManager(val config: KafkaConfig,
   /**
    * Stop the given partitions.
    *
-   * @param partitionsToStop    A map from a topic partition to a boolean indicating
-   *                            whether the partition should be deleted.
+   * @param partitionsToStop                A map from a topic partition to a boolean indicating
+   *                                        whether the partition should be deleted.
+   * @param partitionsMaybeToDeleteRemote   A set of topic partitions that may need to delete
+   *                                        remote segments.
    *
-   * @return                    A map from partitions to exceptions which occurred.
-   *                            If no errors occurred, the map will be empty.
+   * @return                                A map from partitions to exceptions which occurred.
+   *                                        If no errors occurred, the map will be empty.
    */
   protected def stopPartitions(
-    partitionsToStop: Map[TopicPartition, Boolean]
+    partitionsToStop: Map[TopicPartition, Boolean],
```

Comment:
Yes, this pr doesn't support KRaft mode, we could create a new jira to address this issue.

Reply:
Will the KRaft support be added in 3.6.0?

Reply:
Might not.

Reply:
Could you file a jira to track that?

Reply:
We can't commit this without KRaft support. ZK mode is deprecated and we cannot add new features there that don't exist in KRaft.

Reply:
@DL1231 How does it look in regards getting KRaft support added into this PR so it can get merged? We are hitting issues in production that I believe this PR is meant to fix (see https://github.com/aiven/kafka/issues/35).

Alternately if you don't have time to finish it, I can have a look at finishing the PR off so that it can be merged.